src/main/java/com/karol/kindergartenmanagementsystem/model/User.java

```diff
@@ -0,0 +1,73 @@
+package com.karol.kindergartenmanagementsystem.model;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```

Comment:
@Data is like having implicit @Getter, @Setter, @ToString, @EqualsAndHashCode and **@RequiredArgsConstructor** annotations on the class (except that no constructor will be generated if any explicitly written constructors already exist).